tests/helpers/sharingHelper.js

```diff
@@ -0,0 +1,157 @@
+const fetch = require('sync-fetch')
+const {
+  getAuthHeaders,
+  applicationFormUrlEncoded
+} = require('../pactHelper.js')
+
+const shareEndPoint = '/ocs/v1.php/apps/files_sharing/api/v1/shares'
+const publicFilesEndPoint = '/remote.php/dav/public-files'
+
+/**
+ * returns the full and sanitized URL of the dav resource
+ *
+ * @returns {string}
+ */
+const getSharingEndPoint = function () {
+  return (process.env.PROVIDER_BASE_URL + shareEndPoint)
+    .replace(/([^:])\/{2,}/g, '$1/')
+}
+
+/**
+ * returns the full and sanitized URL of public-files
+ *
+ * @returns {string}
+ */
+const getPublicFilesEndPoint = function () {
+  return (process.env.PROVIDER_BASE_URL + publicFilesEndPoint)
+    .replace(/([^:])\/{2,}/g, '$1/')
+}
+
+/**
+ * share a file or folder using webDAV api.
+ *
+ * @param {string} username
+ * @param {string} userPassword
+ * @param {string} path
+ * @param {number} shareType
+ * @param {string} shareWith
+ * @param {number} permissions
+ * @param {string} name
+ * @param {boolean} publicUpload
+ * @param {string} password
+ * @param {Date} expireDate
+ * @returns {*} result of the fetch request
+ */
+const shareResource = function (
+  username,
+  userPassword,
+  path,
+  shareType,
+  shareWith,
+  permissions,
+  name,
+  publicUpload,
+  password,
+  expireDate
+) {
+  const data = validateParams({
+    path, shareType, shareWith, permissions, name, publicUpload, password, expireDate
+  })
+  return fetch(getSharingEndPoint() + '?format=json', {
+    method: 'POST',
+    body: data,
+    headers: {
+      authorization: getAuthHeaders(username, userPassword),
+      ...applicationFormUrlEncoded
+    }
+  })
+}
+
+/**
+ * get shares of a file/folder
+ *
+ * @param {string} username
+ * @param {string} userPassword
+ * @param {string} path
+ * @returns {*} result of the fetch request
+ */
+const getShareInfoByPath = function (username, userPassword, path) {
+  return fetch(getSharingEndPoint() + `?path=${encodeURIComponent(path)}&format=json`, {
+    method: 'GET',
+    headers: {
+      authorization: getAuthHeaders(username, userPassword)
+    }
+  })
+}
+
+/**
+ *
+ * @param {object} params
+ * @returns {string}
+ */
+const validateParams = function (params) {
+  let data = ''
+  if (params) {
+    if (params.path !== undefined && params.path !== null && params.path !== '') {
```

Comment:
```suggestion
    if (params.path) {
```
I think that should do in JS see https://fellowtuts.com/jquery/check-undefined-null-empty-or-blank-variable-in-javascript/